docs/src/snap/explanation/upgrade.md

```diff
@@ -0,0 +1,109 @@
+
+Patch upgrades address bug fixes and are typically safe, introducing no
+breaking changes.
+Patch upgrades do not change the channel selected (for example a patch upgrade
+from stable/1.32.0 to stable/1.32.1). Minor versions upgrades add new
+features or deprecate existing features without breaking changes. These upgrades
+change the version of the channel selected
+(for example a minor upgrade from stable/1.32.x to stable/1.33.x).
+For more information on channels see the [explanation page].
+
+## Sequential upgrades
+
+Upgrades must always be done sequentially. It is not recommended to upgrade by
+more than one minor version at a time (for example upgrading from 1.31 to 1.33
+is not supported). Changing between minor releases (for example upgrading
+from 1.31 to 1.32) is also only recommended when you are on the
+latest patch of the current version (for example 1.31.x where x is the latest
+available patch in 1.31 channel). This is in line with upstream Kubernetes
+[version skew policy].
+
+## Upgrade approach
+
+It is important to upgrade your cluster with the same method you used to install
+the cluster. If you have deployed {{product}} using Juju, you must refresh
+using Juju controller commands. Similarly with snap and CAPI deployments,
+the original install method must be used.
+
+
+## Snap upgrades
+
+Updates for the {{product}} snap are checked several times a day with the
+default [snap refresh fucntionality]. Patch upgrades will be applied
+automatically unless they are manually stopped. Changes applied during the
+latest refresh to the snap can be viewed using the command `snap change`.
+Minor version upgrades need manual intervention and must be carried out on all
+snaps individually. The length of time taken to upgrade the snap will depend
+largely on your local setup, the services running on the cluster, resources
+available to the cluster,the intensity of the workloads etc.
+
+If you would like to manage the upgrading of your cluster using the snap please
+see the how-to guide on [managing cluster upgrades].
+
+## Charm upgrades
+
+Upgrading the charm must be done manually for both patch and minor upgrades on
+a per model basis. To ensure that the revision of the charm always matches the
+revision of the snap it orchestrates, the `refresh` function of the snap has
+been frozen and all upgrades must be done through Juju. The `juju refresh`
+command instructs the Juju controller
+to use the new charm revision within the current charm channel or to upgrade to
+the next channel based on the parameters provided. The charm code
+is simultaneously replaced on each unit. Then, the k8s snap is updated
+unit-by-unit. This is in order to maintain a highly-available kube-api-server
+endpoint, starting with the Juju leader unit for the application. To ensure a
+smooth upgrade process, the pre-upgrade-check should be run before
+all upgrades. This ensures that you cluster is in a stable and ready state for
+an upgrade.
```

Comment:
Perhaps we can rephrase this. The `pre-upgrade-check` is mandatory before upgrading a cluster. In addition to verifying the cluster's health, it also creates the upgrade stack which will be used after the charm is upgraded to orchestrate the upgrade process across the target application (`k8s` or `k8s-worker`) units